pytorch3d/implicitron/dataset/dataset_base.py

```diff
@@ -144,6 +151,30 @@ def __getitem__(self, key):
     def __len__(self):
         return len(fields(self))
 
+    def crop_by_bbox_(self, box_crop_context):
+        clamp_bbox_xyxy = _clamp_box_to_image_bounds_and_round(
+            _get_clamp_bbox(
+                self.bbox_xywh,
+                image_path=self.image.path,
+                box_crop_context=box_crop_context,
+            ),
+            image_size_hw=tuple(self.fg_probability.shape[-2:]),
+        )
+        self.crop_bbox_xywh = _bbox_xyxy_to_xywh(clamp_bbox_xyxy)
+
+        self.fg_probability = _crop_around_box(
+            self.fg_probability, clamp_bbox_xyxy, self.mask_path
+        )
+        self.image_rgb = _crop_around_box(self.image_rgb, clamp_bbox_xyxy, self.image.path)
+
+        depth_bbox_xyxy = _rescale_bbox(clamp_bbox_xyxy, entry.image.size, self.depth_map.shape[-2:])
+        self.depth_map = _crop_around_box(self.depth_map, depth_bbox_xyxy, self.depth_path)
+
+        depth_mask_bbox_xyxy = _rescale_bbox(clamp_bbox_xyxy, entry.image.size, self.depth_mask.shape[-2:])
+        self.depth_mask = _crop_around_box(self.depth_mask, depth_mask_bbox_xyxy, self.mask_path)
+
+        self.camera.principal_point_px -= clamp_bbox_xyxy[:2]
```

Comment:
I don’t think `camera.principal_point_px` is a thing – you probably create a new attribute here that is not used anywhere.
Note that cropping affects both focal length and principal point. Focal length is adjusted based on bounding box extent (see `scale` arg to `_get_pytorch3d_camera`).
Ultimately, the logic we want is: initially, camera should ignore bounding box (so the intrinsics correspond to the uncropped image). When this function is called, the intrinsics are adjusted to match the cropped image (they should account for both scale and shift).
Also, it is safer to re-create the cameras rather than modify in-place since the intrinsics may be stored either as K matrix or as (focal_length, principal_point), and constructors are guaranteed to keep the object consistent, otherwise you have to dig into the implementations.
We can have a call on Friday if we want to know more about the theory.